engine.js

```diff
@@ -0,0 +1,107 @@
+"format cjs";
+
+var wrap = require('word-wrap');
+var objectAssign = require('object-assign');
+var objectValues = require('object-values');
+var forEach = require('lodash.foreach');
+var mapKeys = require('lodash.mapkeys');
+var mapValues = require('lodash.mapvalues');
+var rightPadKeys = require('right-pad-keys');
+var conventionalCommitTypes = require('conventional-commit-types');
+
+// This can be any kind of SystemJS compatible module.
+// We use Commonjs here, but ES6 or AMD would do just
+// fine.
+module.exports = function (options) {
+
+  options = options || {};
+
+  var types = objectAssign({}, options.types || conventionalCommitTypes.types);
+  forEach(types, function (type, key) {
+    type.key = key;
```

Comment:
I don't like that we are modifying variables that are given to us, it's an antipattern. Shouldn't we be able to get away with just one map?

```js
var choices = types.map(type => {
  // ...
})
```